frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx

```diff
@@ -97,23 +214,47 @@ export const OperatorHubList: React.FC<OperatorHubListProps> = ({
             [OperatorHubCSVAnnotationKey.supportWorkflow]: marketplaceSupportWorkflow,
           } = currentCSVAnnotations;
 
+          const subscription = subscriptionFor(subscriptions?.data)(operatorGroups?.data)(
+            pkg.status.packageName,
+          )(namespace);
+
+          const clusterServiceVersion = clusterServiceVersionFor(
+            pkg,
+            clusterServiceVersions?.data,
+            currentCSVDesc?.version,
+          );
```

Comment:
I would update this helper to just accept the CSV name and list of CSV's to search and use `subscription.status.currentCSV` instead of the package manifest default channel CSV:
```suggestion
          const clusterServiceVersion = clusterServiceVersionFor(
            clusterServiceVersions?.data,
            subscription?.status.currentCSV,
          );
```